packages/remix/src/utils/instrumentServer.ts

```diff
@@ -262,8 +272,18 @@ function getTraceAndBaggage(): { sentryTrace?: string; sentryBaggage?: string }
 function makeWrappedRootLoader(origLoader: DataFunction): DataFunction {
   return async function (this: unknown, args: DataFunctionArgs): Promise<Response | AppData> {
     const res = await origLoader.call(this, args);
+    const traceAndBaggage = getTraceAndBaggage();
+
+    // Note: `redirect` and `catch` responses do not have bodies to extract
+    if (isResponse(res) && !isRedirectResponse(res) && !isCatchResponse(res)) {
+      const data = await extractData(res);
+
+      if (typeof data === 'object') {
+        return { ...data, ...traceAndBaggage };
```

Comment:
Do we need to even return `traceAndBaggage` if it's a response? Couldn't we just just return the res?

Reply:
Wait I'm dumb - yes we do, nvm.